app/src/main/java/org/mozilla/fenix/home/Mode.kt

```diff
@@ -50,14 +52,23 @@ class CurrentMode(
 
     private val accountManager by lazy { context.components.backgroundServices.accountManager }
 
+    // The onboarding configuration is retrieved lazily because:
+    // - We do not want to record exposure exposure if a user is not encountering onboarding
+    // - We would like to evaluate the configuration only once (and thus it's kept in memory
+    // and not re-evaluated)
+    private val onboardingConfig by lazy {
```

Comment:
Because a new `CurrentMode` is created in `onCreateView` of `HomeFragment` [here](https://github.com/mozilla-mobile/fenix/blob/3e7d0bf8fbfbcb88eb447d63cf4b74beb14d88af/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt#L242) when a user taps into Signin or the privacy notice then goes back to onboarding screen this will be recreated (inside the new `CurrentMode`) which will cause it to run multiple times. I don't think it's desired that we re-record this exposure but maybe confirm that.

Putting the lazy val config in `OnboardingState` would fix this issue but a user who goes through the sign-in flow will call `recordExposure()` again when they return to the onboarding screen due to the new signed-in state. If this is not desired then this will need to be placed outside of `CurrentMode` and `Mode`.

Reply:
Great catch 🙈 Moved the configuration to `FenixOnboarding` which seemed like the more natural home anyways. The `FenixOnboarding` is a `lazy` member of the `HomeFragment` so I imagine it's only initialized once, but please correct my assumptions (as I've been wrong a couple of times 😆)

Reply:
Ya, this looks great now and a much better place for the config 😄